internal/database/permission_sync_jobs.go

```diff
@@ -0,0 +1,269 @@
+	return &permissionSyncJobStore{Store: txBase}, err
+}
+
+func (s *permissionSyncJobStore) Done(err error) error {
+	return s.Store.Done(err)
+}
+
+func (s *permissionSyncJobStore) CreateUserSyncJob(ctx context.Context, user int32, opts PermissionSyncJobOpts) error {
+	job := &PermissionSyncJob{
+		UserID:           int(user),
+		HighPriority:     opts.HighPriority,
+		InvalidateCaches: opts.InvalidateCaches,
+	}
+	if !opts.NextSyncAt.IsZero() {
+		job.ProcessAfter = opts.NextSyncAt
+	}
+	return s.createSyncJob(ctx, job)
+}
+
+func (s *permissionSyncJobStore) CreateRepoSyncJob(ctx context.Context, repo int32, opts PermissionSyncJobOpts) error {
+	job := &PermissionSyncJob{
+		RepositoryID:     int(repo),
+		HighPriority:     opts.HighPriority,
+		InvalidateCaches: opts.InvalidateCaches,
+	}
+	if !opts.NextSyncAt.IsZero() {
+		job.ProcessAfter = opts.NextSyncAt
+	}
+	return s.createSyncJob(ctx, job)
+}
+
+const permissionSyncJobCreateQueryFmtstr = `
+INSERT INTO permission_sync_jobs (
+	process_after,
+	repository_id,
+	user_id,
+	high_priority,
+	invalidate_caches
+)
+VALUES (
+	%s,
+	%s,
+	%s,
+	%s,
+	%s
+)
+RETURNING %s
+`
+
+func (s *permissionSyncJobStore) createSyncJob(ctx context.Context, job *PermissionSyncJob) error {
+	q := sqlf.Sprintf(
+		permissionSyncJobCreateQueryFmtstr,
+		dbutil.NullTimeColumn(job.ProcessAfter),
+		dbutil.NewNullInt(job.RepositoryID),
+		dbutil.NewNullInt(job.UserID),
+		job.HighPriority,
+		job.InvalidateCaches,
+		sqlf.Join(PermissionSyncJobColumns, ", "),
+	)
+
+	return scanPermissionSyncJob(job, s.QueryRow(ctx, q))
+}
+
+type ListPermissionSyncJobOpts struct {
+	ID     int
+	UserID int
+	RepoID int
+}
+
+func (opts ListPermissionSyncJobOpts) sqlConds() []*sqlf.Query {
+	conds := []*sqlf.Query{}
+
+	if opts.ID != 0 {
+		conds = append(conds, sqlf.Sprintf("id = %s", opts.ID))
+	}
+	if opts.UserID != 0 {
+		conds = append(conds, sqlf.Sprintf("user_id = %s", opts.UserID))
+	}
+	if opts.RepoID != 0 {
+		conds = append(conds, sqlf.Sprintf("repository_id = %s", opts.RepoID))
+	}
+
+	return conds
+}
+
+func (s *permissionSyncJobStore) List(ctx context.Context, opts ListPermissionSyncJobOpts) ([]*PermissionSyncJob, error) {
+	conds := opts.sqlConds()
+
+	var whereClause *sqlf.Query
+	if len(conds) != 0 {
+		whereClause = sqlf.Sprintf("WHERE %s", sqlf.Join(conds, "\n AND "))
+	} else {
+		whereClause = sqlf.Sprintf("")
+	}
+
+	q := sqlf.Sprintf(
+		listPermissionSyncJobQueryFmtstr,
+		sqlf.Join(PermissionSyncJobColumns, ", "),
+		whereClause,
+	)
+
+	rows, err := s.Query(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { err = basestore.CloseRows(rows, err) }()
+
+	var syncJobs []*PermissionSyncJob
+	for rows.Next() {
+		job, err := ScanPermissionSyncJob(rows)
+		if err != nil {
+			return nil, err
+		}
+		syncJobs = append(syncJobs, job)
+	}
+
+	return syncJobs, nil
+}
+
+const listPermissionSyncJobQueryFmtstr = `
+SELECT %s
+FROM permission_sync_jobs
+%s -- whereClause
```

Comment:
nit: "whereClause" comment will be after the clause itself, probably it can be moved one line above for better readability?

```suggestion
-- whereClause
%s
```

Reply:
eh. I mostly put it there for readers of this string, not for someone looking at the SQL query, so I think it's fine for now